torch_xla/experimental/stablehlo_saved_model.py

```diff
@@ -137,3 +143,257 @@ def export_torch_model(model: torch.nn.Module,
   # mistakenlly update the input tensors.
   torch_xla._XLAC._clear_pending_irs(str(xm.xla_device()))
   return stablehlo_model
+
+
+class VariableType(enum.Enum):
+  INPUT_ARG = 'input_arg'
+  PARAMETER = 'parameter'
+  CONSTANT = 'constant'
+
+
+@dataclass
+class VariableSignature:  # either argument or parameters
+  shape: List[int]
+  dtype: str
+
+
+@dataclass
+class InputLocation:
+  type_: VariableType
+  position: int = -1
+  name: str = ''
+
+  @classmethod
+  def parameter(cls, name: str):
+    return cls(type_=VariableType.PARAMETER, name=name)
+
+  @classmethod
+  def input_arg(cls, position: int):
+    return cls(type_=VariableType.INPUT_ARG, position=position)
+
+  @classmethod
+  def constant(cls, position):
+    return cls(type_=VariableType.CONSTANT, position=position)
+
+
+@dataclass
+class StableHLOFunctionMeta:
+  # name of the callable.
+  name: str
+  # version.
+  stablehlo_version: int
```

Comment:
Not sure if we should use a str? Seems StableHLO version is major.minor.patch. https://github.com/openxla/stablehlo/blob/bf4ff668528f789bd12480de2dfbe68cab73dcc0/stablehlo/dialect/Version.h#L34